### PR TITLE
chore(Breadcrumbs): add generic parameter <T extends BreadcrumbsItem>

### PR DESCRIPTION
**Pull request by ma-efremoff**

### CODE REVIEW

File: src/components/Breadcrumbs/BreadcrumbsItem.tsx [hunk 1]
```diff
@@ -3,43 +3,43 @@ import {Link} from '../Link';
...
-    data,
-    isCurrent,
-    isPrevCurrent,
-    renderItem,
-}: Props) {
-    const {text, href, action} = data;
+export class BreadcrumbsItem<
+    T extends IBreadcrumbsItem = IBreadcrumbsItem,
+> extends React.PureComponent<Props<T>> {
+    render() {
```
Comment (ogonkov) [inline src/components/Breadcrumbs/BreadcrumbsItem.tsx:20]: Why you convert FC component to class?

Comment (ma-efremoff) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:20]: It is impossible to create generic components with React.memo

**Review by vrozaev — APPROVED** [review]

File: src/components/Breadcrumbs/BreadcrumbsItem.tsx (hunk 1 above)
Comment (ogonkov) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:20]: It is possible https://stackoverflow.com/a/60170425

File: src/components/Breadcrumbs/BreadcrumbsItem.tsx [hunk 2]
```diff
@@ -3,21 +3,23 @@ import {Link} from '../Link';
...
 }
 
 const b = block('breadcrumbs');
 
-export const BreadcrumbsItem = React.memo(function BreadcrumbsItem({
+function BreadcrumbsItem<T extends IBreadcrumbsItem = IBreadcrumbsItem>({
     data,
     isCurrent,
     isPrevCurrent,
     renderItem,
```
Comment (ogonkov) [inline src/components/Breadcrumbs/BreadcrumbsItem.tsx:21]: Isn't it enough to make function generic?

```suggestion
export const BreadcrumbsItem = React.memo(function BreadcrumbsItem<T extends IBreadcrumbsItem = IBreadcrumbsItem>({
    data,
    isCurrent,
    isPrevCurrent,
    renderItem,
```

Comment (ma-efremoff) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:21]: Yes, it is. But the last version has less changed rows. But if you insist I will bring back the version with function.

Comment (ogonkov) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:21]: There is also `displayName` disappear, and i personally like named instead of `default`

Comment (ma-efremoff) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:21]: Please suggest your fix for the file.

Comment (ogonkov) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:21]: I suggest (above) to just add generic to function and keep everything other unchanged

```jsx
export const BreadcrumbsItem = React.memo(function BreadcrumbsItem<T extends IBreadcrumbsItem = IBreadcrumbsItem>({
    data,
    isCurrent,
    isPrevCurrent,
    renderItem,
}: Props<T>) {
    //
});

BreadcrumbsItem.displayName = 'Breadcrumbs.Item';
```

Comment (ma-efremoff) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:21]: Please provide full solution, your one has no any generic-component, and there is nothing you can use for type-cast as described in link above.

**Review by ogonkov — APPROVED** [review]

**Pull request merged**